### PR TITLE
[TM-132] Confirmation Page for Declined Funds (Receiver)

### PR DESCRIPTION
**Pull request by mmiddleton3301**
The confirmation page for declined funds, from the receiver's perspective.

Note: targets `TM-54-Decline-Funding-For-Approved-Applications`.

See also:

* https://github.com/SkillsFundingAgency/das-apim-endpoints/pull/532

### CODE REVIEW

File: src/SFA.DAS.LevyTransferMatching.Infrastructure/Services/ApplicationsService/ApplicationsService.cs [hunk 1]
```diff
@@ -73,5 +73,25 @@ public async Task<GetAcceptedResponse> GetAccepted(long accountId, int applicati
...
+        public async Task<GetDeclinedResponse> GetDeclined(long accountId, int applicationId)
+        {
+            var response = await _httpClient.GetAsync($"accounts/{accountId}/applications/{applicationId}/declined");
+
+            GetDeclinedResponse getDeclinedResponse = null;
+            if (response.IsSuccessStatusCode)
+            {
+                getDeclinedResponse = JsonConvert.DeserializeObject<GetDeclinedResponse>(await response.Content.ReadAsStringAsync());
+            }
+            else
```
[the remark this answers is not in the document]
Comment (mmiddleton3301) [reply src/SFA.DAS.LevyTransferMatching.Infrastructure/Services/ApplicationsService/ApplicationsService.cs:86]: Good spot 👀; changed.

File: src/SFA.DAS.LevyTransferMatching.Web.UnitTests/Controllers/ApplicationsControllerTests.cs [hunk 2]
```diff
@@ -153,5 +153,45 @@ public void POST_Application_SelectedActionIsDecline_ThrowsNotImplementedExcepti
             // Act/Assert
             Assert.ThrowsAsync<System.NotImplementedException>(() => _controller.Application(request));
         }
+
+        [Test]
+        public async Task GET_Declined_ApplicationExists_ReturnsViewAndModel()
+        {
+            // Arrange
+            var request = _fixture.Create<DeclinedRequest>();
+            _orchestrator
```
[the remark this answers is not in the document]
Comment (mmiddleton3301) [reply src/SFA.DAS.LevyTransferMatching.Web.UnitTests/Controllers/ApplicationsControllerTests.cs:170]: Good spot; to be honest, I'm not sure what I was doing previously, by simply passing in the same instance. It was meant to be an `It.Is` check. I did this all over the same test class; so I've fixed this all over, not just on this test.

File: src/SFA.DAS.LevyTransferMatching.Web.UnitTests/Controllers/ApplicationsControllerTests.cs [hunk 3]
```diff
@@ -153,5 +153,45 @@ public void POST_Application_SelectedActionIsDecline_ThrowsNotImplementedExcepti
...
+            Assert.NotNull(declinedViewModel);
+        }
+
+        [Test]
+        public async Task GET_Declined_ApplicationDoesntExist_ReturnsNotFound()
+        {
+            // Arrange
+            var request = _fixture.Create<DeclinedRequest>();
+            _orchestrator
+                .Setup(x => x.GetDeclinedViewModel(request))
```
[the remark this answers is not in the document]
Comment (mmiddleton3301) [reply src/SFA.DAS.LevyTransferMatching.Web.UnitTests/Controllers/ApplicationsControllerTests.cs:193]: See comment above.

File: src/SFA.DAS.LevyTransferMatching.Web/Views/Applications/Declined.cshtml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        <div class="govuk-grid-row">
+            <div class="govuk-grid-column-two-thirds">
+
+                <div class="govuk-panel govuk-panel--confirmation application-complete govuk-!-margin-bottom-9">
+                    <h1 class="govuk-panel__title">You have successfully declined a transfer of funds</h1>
+                </div>
+
+                <h2 class="govuk-heading-m ">What happens next</h2>
+
+                <p>Your transfer of funds from <strong>@Model.EmployerNameAndReference</strong> has been declined. You can apply to another opportunity.</p>
```
[the remark this answers is not in the document]
Comment (mmiddleton3301) [reply src/SFA.DAS.LevyTransferMatching.Web/Views/Applications/Declined.cshtml:20]: Thanks - that looks a little more like the screenshot on the ticket now.

File: src/SFA.DAS.LevyTransferMatching.Web/Views/Applications/Declined.cshtml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+                <p>
+                    <br />
```
[the remark this answers is not in the document]
Comment (mmiddleton3301) [reply src/SFA.DAS.LevyTransferMatching.Web/Views/Applications/Declined.cshtml:23]: Thanks. I've had to use `govuk-!-margin-top-4`, but it looks pretty close to the screenshot on the ticket now.

**Review by sscaife — APPROVED** [review]

**Pull request merged**